src/linalg.jl

```diff
@@ -86,3 +86,17 @@ function Base.getproperty(C::Cholesky{T, <:BlockDiagonal{T}}, x::Symbol) where T
     end
     return BlockDiagonal(map(f, blocks(B)))
 end
+
+# 3-Argument mul!
+LinearAlgebra.mul!(C::BlockDiagonal, A::BlockDiagonal, B::BlockDiagonal) = LinearAlgebra.mul!(C, A, B, true, false)
```

Comment:
since 5-arg `mul!` requires Julia v1.3, I think we need to wrap these two methods in an `if` block

e.g. like 
```suggestion
if VERSION ≥ v"1.3"
    LinearAlgebra.mul!(C::BlockDiagonal, A::BlockDiagonal, B::BlockDiagonal) = LinearAlgebra.mul!(C, A, B, true, false)
```
(we could find out the exact `VERSION` in which `mul!` was added, but i'm not sure it's worth it)

And similarly for the corresponding tests.